_docs/cardano/2017-01-02-blocks.md

```diff
@@ -1,17 +1,62 @@
 ---
 layout: default
-title:  Handling Blocks and Headers
-permalink: /protocols/handling-blocks-and-headers/
-group: protocols
+title:  Blocks in Cardano SL
+permalink: /cardano/blocks/
+group: cardano
 ---
 
 [//]: # (Reviewed at 60033350e60408fc79f202491e6985b3b47acd90)
 
-# Handling Blocks and Headers
+# Blocks in Cardano SL
 
-This guide describes blocks handling logic, which is defined in [Pos.Block.Logic](https://github.com/input-output-hk/cardano-sl/blob/517a72801c0bbb11a34c8d6a6d528fff5f094471/src/Pos/Block/Logic.hs) module.
+This guide describes blocks design and logic of the block handling.
 
-## Concepts
+Block-related types are defined in [Pos.Types.Block.Types](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Types/Block/Types.hs) module. Logic is defined in [Pos.Block.Logic](https://github.com/input-output-hk/cardano-sl/blob/517a72801c0bbb11a34c8d6a6d528fff5f094471/src/Pos/Block/Logic.hs) module.
+
+## Design
+
+Block is a fundamental part of the ledger. There are two kinds of blocks: **main** block and **genesis** block.
+
+### Main Block
+
+Main block consists of *header* and *body*. A header contains meta-information about the block,
+like a block signature. A body contains 4 primary components:
+
+1. transactions (with a separate witnesses for them),
+2. SSC,
+3. delegation,
+4. update system.
+
+**Transactions** are stored in the Merkle tree. The list of witnesses for each transaction is stored in the block separately.
+Please read about [Transactions in Cardano SL](/cardano/transactions/) for more info about transaction and witnesses.
+
+**SSC** (Shared Seed Computation) is used for the [Follow-the-Satoshi algorithm](/cardano/proof-of-stake/#follow-the-satoshi). 
+Every epoch we must elect slot-leaders for the next epoch, these slot-leaders will be able to generate new main blocks and add them in the ledger. So we use SSC in the election process. 
+
+**Delegation** is a mechanism that allows a node to be a delegate of some other node. For example, if some node `D` is a delegate of the node `I`, then:
+
+* if `I` is a leader, node `D` have a right to issue a new block;
+* if `I` has some money, we can think that `D` owns these money.
+
+Delegate can prove his rights with a special certificate.
+
+Special payload is using for the **update system** mechanism. It contains a proposal for software update (update for the 
+nodes and update of the protocol itself) and a vote for the particular update (stakeholders can approve or reject one). 
+Please read about [Update System Model](/cardano/update-mechanism/) for more info.
+
+### Genesis Block
+
+Genesis block doesn't contain transactions, and we have just one genesis block for each epoch. Genesis block has a header and a body as well as main block. Block's body contains:
+
+1. an index of the epoch this block associated with,
+2. the list of stakeholders for this epoch,
+3. chain difficulty.
```

Comment:
I think it's better not to use words `chain difficulty` in documentation. Prefer `chain height`.